src/discoveryTools.ts

```diff
@@ -65,59 +167,60 @@ function getSChainNodesCount( joSChainNetworkInfo ) {
     }
 }
 
-export function isSChainNodeFullyDiscovered( joNode ) {
-    if( ! joNode )
+export function isSChainNodeFullyDiscovered( joNode: TSChainNode ): boolean {
+    if( !joNode )
         return false;
-    if( joNode && "imaInfo" in joNode && typeof joNode.imaInfo == "object" &&
-        "t" in joNode.imaInfo && typeof joNode.imaInfo.t == "number" &&
+    if( joNode && "imaInfo" in joNode && typeof joNode.imaInfo === "object" &&
```

Comment:
Why do you need to check types here (and in other such cases)?

Reply:
This is fullest and safest validation of JSON data came from `skaled`, we need it, it was pain to came to this, but this is long tested and works